### PR TITLE
#20825 Missing required argument $productAvailabilityChecks of Magent…

### PR DESCRIPTION
**Pull request by kisroman**
…o\Sales\Model\Order\Reorder\OrderedProductAvailabilityChecker.

### Fixed Issues (if relevant)

1. https://github.com/magento/magento2/issues/20825

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @kisroman. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/Sales/etc/di.xml [hunk 1]
```diff
@@ -1015,4 +1015,11 @@
     <preference
             for="Magento\Sales\Api\OrderCustomerDelegateInterface"
             type="Magento\Sales\Model\Order\OrderCustomerDelegate" />
+    <type name="Magento\Sales\Model\Order\Reorder\OrderedProductAvailabilityChecker">
+        <arguments>
+            <argument name="productAvailabilityChecks" xsi:type="array">
+                <item name="configurable" xsi:type="object">Magento\ConfigurableProductSales\Model\Order\Reorder\OrderedProductAvailabilityChecker</item>
+            </argument>
+        </arguments>
+    </type>
```
Comment (dmytro-ch) [inline app/code/Magento/Sales/etc/di.xml:1022]: Unfortunately, the current solution is not resolving the dependency issue between the modules.
I guess this definition still should be kept under **Magento_ConfigurableProductSales** (in `app/code/Magento/ConfigurableProductSales/etc/di.xml`).

And it should be enough to inject an empty array as default value of `productAvailabilityChecks` parameter (in `app/code/Magento/Sales/etc/di.xml`).
```
<type name="Magento\Sales\Model\Order\Reorder\OrderedProductAvailabilityChecker">
    <arguments>
        <argument name="productAvailabilityChecks" xsi:type="array" />
    </arguments>
</type>
```

**Review by dmytro-ch — CHANGES_REQUESTED** [review]
Hi @kisroman,
thank you for your contribution!
Could you please check the code review comment?

File: app/code/Magento/Sales/etc/di.xml (hunk 1 above)
Comment (kisroman) [reply app/code/Magento/Sales/etc/di.xml:1022]: Fixed. Check pls.

**Review by dmytro-ch — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @dmytro-ch, thank you for the review.
[ENGCOM-4543](https://jira.corp.magento.com/browse/ENGCOM-4543) has been created to process this Pull Request

**Comment by soleksii** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**